pkg/reconciler/gc/v2/gc.go

```diff
@@ -66,15 +66,16 @@ func Collect(
 	})
 
 	// Delete stale revisions while more than min remain
-	remaining := len(revs)
-	nonstale := make([]*v1.Revision, 0, remaining)
+	count, remaining := 0, len(revs)
+	nonstale := make([]*v1.Revision, remaining)
```

Comment:
This doesn't change the memory usage since the backing array will still be of `cap(remaining)`

Similar to below - I wonder if partitioning the array (ie. `[non-active,stale, active]`) would make processing easier and allow you to drop the need for this slice - I haven't thought deeply about